files/templates/scrapeconfigs/additional-scrape-configs.template.yaml

```diff
@@ -527,36 +548,30 @@
   - source_labels: [__meta_kubernetes_service_annotationpresent_giantswarm_io_monitoring, __meta_kubernetes_service_labelpresent_giantswarm_io_monitoring]
     regex: .*(true).*
     action: keep
-  - source_labels: [__meta_kubernetes_service_labelpresent_operated_prometheus]
-    regex: (true)
-    action: drop
-  - source_labels: [__meta_kubernetes_service_annotationpresent_giantswarm_io_monitoring_port]
-    regex: (true)
-    action: keep
-  - source_labels: [__meta_kubernetes_service_annotationpresent_giantswarm_io_monitoring_path]
-    regex: (true)
-    action: keep
-  - regex: (.*)
-    target_label: __address__
-    replacement: [[ .APIServerURL ]]:443
+    # if __meta_kubernetes_service_annotation_giantswarm_io_monitoring_path is present, we use it as the metrics path
+  - source_labels: [__meta_kubernetes_service_annotation_giantswarm_io_monitoring_path]
+    action: replace
+    target_label: __metrics_path__
+    regex: (.+)
+    # if __meta_kubernetes_service_annotation_giantswarm_io_monitoring_port, we use it as the metrics port
+  - source_labels: [__address__, __meta_kubernetes_service_annotation_giantswarm_io_monitoring_port]
     action: replace
-  - source_labels: [__meta_kubernetes_namespace, __meta_kubernetes_pod_name, __meta_kubernetes_service_annotation_giantswarm_io_monitoring_port, __meta_kubernetes_service_annotation_giantswarm_io_monitoring_path]
-    regex: (.*);(.*);(.*);(.*)
+    target_label: __address__
+    regex: ([^:]+)(?::\d+)?;(\d+)
+    replacement: $1:$2
+  - source_labels: [__meta_kubernetes_namespace, __meta_kubernetes_pod_name, __address__, __metrics_path__]
+    regex: (.*);(.*);(?:[^:]+:)(\d+);(.*)
```

Comment:
Same remark, I believe we could have something simpler like `(.*);(.*);.+:(\d+);(.*)`

Reply:
We could, this regex just drops the unused address using ?:. This is merely a convenient way to use regexes :D

Reply:
You can check https://stackoverflow.com/a/3513858 for details on non capturing groups